aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -211,13 +288,75 @@ impl BaseAnonCreds for IndyCredxAnonCreds {
         sig_type: Option<&str>,
         config_json: &str,
     ) -> VcxCoreResult<(String, String)> {
-        let _ = (issuer_did, schema_json, tag, sig_type, config_json);
-        Err(unimplemented_method_err("credx issuer_create_and_store_credential_def"))
+        let issuer_did = issuer_did.to_owned().into();
+        let schema = serde_json::from_str(schema_json)?;
+        let sig_type = sig_type.map(serde_json::from_str).unwrap_or(Ok(SignatureType::CL))?;
+        let config = serde_json::from_str(config_json)?;
+
+        let schema_seq_no = match &schema {
+            Schema::SchemaV1(s) => s.seq_no,
+        };
+
+        let cred_def_id =
+            credx::issuer::make_credential_definition_id(&issuer_did, schema.id(), schema_seq_no, tag, sig_type)?;
+
+        // If cred def already exists, return it
+        if let Ok(cred_def) = self.get_wallet_record_value(CATEGORY_CRED_DEF, &cred_def_id.0).await {
+            return Ok((cred_def_id.0, cred_def));
+        }
+
+        // Otherwise, create cred def
+        let (cred_def, cred_def_priv, cred_key_correctness_proof) =
+            credx::issuer::create_credential_definition(&issuer_did, &schema, tag, sig_type, config)?;
+
+        let str_cred_def = serde_json::to_string(&cred_def)?;
+
+        // Store stuff in wallet
+        self.wallet
+            .add_wallet_record(CATEGORY_CRED_DEF, &cred_def_id.0, &str_cred_def, None)
+            .await?;
+
+        let str_cred_def_priv = serde_json::to_string(&cred_def_priv)?;
+
+        self.wallet
+            .add_wallet_record(CATEGORY_CRED_DEF_PRIV, &cred_def_id.0, &str_cred_def_priv, None)
+            .await?;
+
+        let str_cred_key_proof = serde_json::to_string(&cred_key_correctness_proof)?;
+
+        self.wallet
+            .add_wallet_record(CATEGORY_CRED_KEY_PROOF, &cred_def_id.0, &str_cred_key_proof, None)
+            .await?;
+
+        self.wallet
+            .add_wallet_record(CATEGORY_CRED_SCHEMA, schema.id(), schema_json, None)
+            .await?;
+
+        let str_schema_id = serde_json::to_string(schema.id())?;
+
+        self.wallet
+            .add_wallet_record(CATEGORY_CRED_SCHEMA_ID, &cred_def_id.0, &str_schema_id, None)
+            .await?;
+
+        // Return the ID and the cred def
+        Ok((cred_def_id.0.to_owned(), str_cred_def))
     }
 
     async fn issuer_create_credential_offer(&self, cred_def_id: &str) -> VcxCoreResult<String> {
-        let _ = cred_def_id;
-        Err(unimplemented_method_err("credx issuer_create_credential_offer"))
+        let cred_def = self.get_wallet_record_value(CATEGORY_CRED_DEF, cred_def_id).await?;
+
+        let correctness_proof = self
+            .get_wallet_record_value(CATEGORY_CRED_KEY_PROOF, cred_def_id)
+            .await?;
+
+        let schema_id = self
+            .get_wallet_record_value(CATEGORY_CRED_SCHEMA_ID, cred_def_id)
+            .await?;
+
+        // If cred_def contains schema ID, why take it as an argument here...?
```

Comment:
I agree. Note that ACA-py does not do this weird `CATEGORY_CRED_SCHEMA_ID` record id mapping stuff. They just get the `schema_id` from the cred_def;

https://github.com/hyperledger/aries-cloudagent-python/blob/main/aries_cloudagent/indy/credx/issuer.py#L179

Reply:
kk fair enough